src/inc/msquic.h

```diff
@@ -195,10 +195,31 @@ typedef enum QUIC_DATAGRAM_SEND_STATE {
 #define QUIC_DATAGRAM_SEND_STATE_IS_FINAL(State) \
     ((State) >= QUIC_DATAGRAM_SEND_LOST_DISCARDED)
 
+//
+// Event type for worker thread starting and stopping.
+//
+typedef enum QUIC_WORKER_EVENT_TYPE {
+    QUIC_WORKER_STARTED = 0,
+    QUIC_WORKER_STOPPED = 1
+} QUIC_WORKER_EVENT_TYPE;
+
+//
+// Callback handler for worker thread starting and stopping.
+//
+typedef
+_IRQL_requires_max_(PASSIVE_LEVEL)
+_Function_class_(QUIC_WORKER_CALLBACK)
+void
+(QUIC_API QUIC_WORKER_CALLBACK)(
+    _In_ QUIC_WORKER_EVENT_TYPE EventType
+    );
+
+typedef QUIC_WORKER_CALLBACK *QUIC_WORKER_CALLBACK_HANDLER;
 
 typedef struct QUIC_REGISTRATION_CONFIG { // All fields may be NULL/zero.
     const char* AppName;
     QUIC_EXECUTION_PROFILE ExecutionProfile;
+    QUIC_WORKER_CALLBACK_HANDLER WorkerStateHandler;
```

Comment:
You can make this non-breaking by putting it behind a "custom" execution profile flag. Though, still I don't believe it should be architected this way.